internal/controller/accessrequest_controller.go

```diff
@@ -479,9 +568,55 @@ func (r *AccessRequestReconciler) handleFinalizer(ctx context.Context, ar *api.A
 	return true, nil
 }
 
+// findObjectsForRoleTemplate will retrieve all AccessRequest resources referencing
+// the given roleTemplate and build a list of reconcile requests to be sent to the
+// controller. Only non-concluded AccessRequests will be added to the reconciliation
+// list. An AccessRequest is defined as concluded if their status is Expired or Denied.
+func (r *AccessRequestReconciler) findObjectsForRoleTemplate(ctx context.Context, roleTemplate client.Object) []reconcile.Request {
+	attachedAccessRequests := &api.AccessRequestList{}
+	listOps := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(roleTemplateField, roleTemplate.GetName()),
+		Namespace:     roleTemplate.GetNamespace(),
+	}
+	err := r.List(ctx, attachedAccessRequests, listOps)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := make([]reconcile.Request, len(attachedAccessRequests.Items))
+	for i, item := range attachedAccessRequests.Items {
+		if !isConcluded(&item) {
+			requests[i] = reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      item.GetName(),
+					Namespace: item.GetNamespace(),
+				},
+			}
+		}
+	}
+	if len(requests) == 0 {
+		return nil
+	}
+	return requests
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *AccessRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().
+		IndexField(context.Background(), &api.AccessRequest{}, roleTemplateField, func(rawObj client.Object) []string {
+			// Extract the ConfigMap name from the ConfigDeployment Spec, if one is provided
```

Comment:
Can you clarify this comment and what the function is doing.

Reply:
done